spanner/sessionclient.go

```diff
@@ -230,11 +254,39 @@ func (sc *sessionClient) executeBatchCreateSessions(client *vkit.Client, createC
 			consumer.sessionCreationFailed(ToSpannerError(ctx.Err()), remainingCreateCount)
 			break
 		}
+		var mdForGFELatency metadata.MD
 		response, err := client.BatchCreateSessions(ctx, &sppb.BatchCreateSessionsRequest{
 			SessionCount:    remainingCreateCount,
 			Database:        sc.database,
 			SessionTemplate: &sppb.Session{Labels: labels},
-		})
+		}, gax.WithGRPCOptions(grpc.Header(&mdForGFELatency)))
+
+		if GFELatencyMetricsEnabled && mdForGFELatency != nil {
+			_, instance, database, errGFE := parseDatabaseName(sc.database)
+			if errGFE != nil {
+				trace.TracePrintf(ctx, nil, "Error getting instance and database name: %v", errGFE)
+				consumer.sessionCreationFailed(ToSpannerError(errGFE), remainingCreateCount)
+				break
+			}
+			// Errors should not prevent initializing the session pool.
+			ctxGFE, errGFE := tag.New(ctx,
+				tag.Upsert(tagKeyClientID, sc.id),
+				tag.Upsert(tagKeyDatabase, database),
+				tag.Upsert(tagKeyInstance, instance),
+				tag.Upsert(tagKeyLibVersion, version.Repo),
+			)
+			if errGFE != nil {
+				trace.TracePrintf(ctx, nil, "Error in adding tags in BatchCreateSessions for GFE Latency: %v", errGFE)
+				consumer.sessionCreationFailed(ToSpannerError(errGFE), remainingCreateCount)
+				break
```

Comment:
This can be removed if you move the error check for the session creation to above this entire block, as it means that we at this point will know that the session creation succeeded and only the extraction of the latency failed. Logging the error from that should be enough, we don't need to report to the client that the creation itself failed (as it didn't in this case).

Reply:
Do you mean we can simply add a trace but don't need to actually break the fxn in case of GFE Latency error?

Reply:
Yes, I think that would be reasonable. It is not an error that the user can do anything with, and the session creation itself succeeded.